src/basic/process-util.c

```diff
@@ -132,6 +132,26 @@ int get_process_comm(pid_t pid, char **ret) {
         return 0;
 }
 
+int pidref_get_comm(const PidRef *pid, char **ret) {
+        _cleanup_free_ char *comm = NULL;
+        int r;
+
+        if (!pidref_is_set(pid))
+                return -ESRCH;
+
+        r = pid_get_comm(pid->pid, &comm);
+        if (r < 0)
+                return r;
+
+        r = pidref_verify(pid);
+        if (r < 0)
+                return r;
+
+        if (ret)
```

Comment:
actually i was thinking we should make these things all optional, i.e. i systematically made the stuff unnecessary for all new stuff i added. we so often start calling these functions for validation purposes of some kind, so that sooner or later we added to a bunch of them the ability to pass NULL as return value, hence I think it makes sense to just start out like that for new code.

I mean I can't tell you where precisely pid_get_comm() actually makes sense to call with a NULL return param, but I couldn't think of a case where safe_atou*() taking NULL makes sense, but still we eventually added cases where it does.

Hence I think in particular for these simple, fundamental calls we should just start out with optional return parameters.